### PR TITLE
Test correct intent slot sanitization in story visualization

### PR DESCRIPTION
**Pull request by gausie**
**Proposed changes**:
- Tests previously didn't look at the contents of a generated DOT file. This test passes, but would be good to check for into the future. 

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: tests/core/test_visualization.py [hunk 1]
```diff
@@ -90,12 +90,13 @@ def test_common_action_prefix_unequal():
...
 
-    generated_graph = nx_pydot.to_pydot(generated_graph)
+    generated_graph = str(nx_pydot.to_pydot(generated_graph))
 
     assert isfile(out_file)
 
     with open(out_file, "r") as graph_file:
         content = graph_file.read()
 
+    assert r'label="/average_weather\{\"season\":\ \"winter\"\}"' in str(generated_graph)
```
Comment (wochinge) [inline tests/core/test_visualization.py]: you did that already above
```suggestion
    assert r'label="/average_weather\{\"season\":\ \"winter\"\}"' in generated_graph
```

Comment (gausie) [reply tests/core/test_visualization.py]: Yeah I originally made this two tests and left that when combining them

File: rasa/core/training/visualization.py [hunk 2]
```diff
@@ -359,9 +359,26 @@ def _create_graph(fontsize: int = 12) -> "networkx.MultiDiGraph":
     return graph
 
 
-def sanitize(s):
+def escape(pattern: Text):
```
Comment (wochinge) [inline rasa/core/training/visualization.py]:
```suggestion
def escape(pattern: Text) -> Text:
```

File: rasa/core/training/visualization.py [hunk 3]
```diff
@@ -359,9 +359,26 @@ def _create_graph(fontsize: int = 12) -> "networkx.MultiDiGraph":
...
+
+    Used to fix different behavior in re.escape() for python
+    versions < 3.7 compared to >= 3.7.
+    """
+    import sys
+    import re
+
+    _special_chars_map = {i: "\\" + chr(i) for i in b"()[]{}?*+-|^$\\.&~# \t\n\r\v\f"}
+
+    if sys.version_info >= (3, 7):
```
Comment (wochinge) [inline rasa/core/training/visualization.py]: please add a comment why you are doing this

File: rasa/core/training/visualization.py [hunk 4]
```diff
@@ -359,9 +359,26 @@ def _create_graph(fontsize: int = 12) -> "networkx.MultiDiGraph":
...
+        return re.escape(pattern)
+    else:
+        return pattern.translate(_special_chars_map)
+
+
+def sanitize(s: Optional[Text] = None):
```
Comment (wochinge) [inline rasa/core/training/visualization.py]:
```suggestion
def sanitize(s: Optional[Text] = None) -> Optional[Text]:
```

File: rasa/core/training/visualization.py [hunk 5]
```diff
@@ -359,9 +359,26 @@ def _create_graph(fontsize: int = 12) -> "networkx.MultiDiGraph":
...
     if s:
-        return re.escape(s)
+        return escape(s)
     else:
         return s
```
Comment (wochinge) [inline rasa/core/training/visualization.py]: I think that makes it a bit clearer what happens here
```suggestion
        return None
```

File: rasa/core/training/visualization.py (hunk 3 above)
Comment (gausie) [reply rasa/core/training/visualization.py]: it explains in the docstring for the function

**Comment by erohmensing** [comment]
@gausie do we still want to merge the test but not the changes to viz?

**Pull request closed without merging**